### ISSUE TITLE
Data size changed

### ISSUE THREAD

**Issue by PFloyd0** [jpmml/sklearn2pmml#320]
Sorry
There are some errors when I  convert my model into pmml.
It seems the data size has been changed but I do not know why.
<img width="510" alt="WeChat Screenshot_20220401112342" src="https://user-images.githubusercontent.com/73516559/161189553-488f6fb4-874e-48f4-954e-5994b90c9630.png">

Thank you!

**Comment by vruusmann** [comment]
The Java stack trace indicates that the K-neighbors regressor object inconsistency is detected on line 71 of `KNeighborsUtil.java`.

This does not match the latest state of the SkLearn2PMML/JPMML-SkLearn stack (ie. should be line 65):
https://github.com/jpmml/jpmml-sklearn/blob/1.7.1/pmml-sklearn/src/main/java/sklearn/neighbors/KNeighborsUtil.java#L65

In other words, please upgrade your SkLearn2PMML package version to the latest (should be 0.78.1), and re-run your experiment.

**Comment by PFloyd0** [comment]
<img width="502" alt="Capture" src="https://user-images.githubusercontent.com/73516559/161461719-fee45887-2f98-4446-9710-e12c67fefa59.PNG">
<img width="302" alt="Capture2" src="https://user-images.githubusercontent.com/73516559/161461722-9addfcfc-fdc4-4bdd-ac37-de175c051d24.PNG">
Hello,
    I have already updated my sklearn2pmml package from 0.78.0 to 0.78.1 but this problem still occurs. 
    Look forward to your reply. Thank you!

**Comment by vruusmann** [comment]
@PFloyd0 Take a look at the Java exception stack trace that you just posted - it still points to line 71.

It means that your SkLearn2PMML package update didn't work.

**Comment by vruusmann** [comment]
> It means that your SkLearn2PMML package update didn't work.

@PFloyd0 My bad, I'm taking back the above comment.

The SkLearn2PMML package is currently based on JPMML-SkLearn 1.6.X codebase (not the recent 1.7.X codebase), so we use legacy line numbers:
https://github.com/jpmml/jpmml-sklearn/blob/1.6.X/src/main/java/sklearn/neighbors/KNeighborsUtil.java#L71

**Comment by vruusmann** [comment]
In short, there is a mismatch between `KNeighborsRegressor._fit_X` shape (reports 60'000 instances) and `KNeightborsRegressor._y` attributes (reports 30'000 instances).

Can you print the values of these two attributes, and see if you see the same mismatch in Python environment.

Second, what are your `offline_rss` and `offline_location` data matrix types? They don't seem to be Pandas data matrix types (`pandas.DataFrame` and `pandas.Series`, respectively), because the `PMMLPipeline.fit(X, y)` method is unable to extract feature and target names.

Are they raw Numpy arrays?

TLDR: What gets printed to console?
```python
print(knn_reg._fit_X.shape)
print(knn_reg._y.shape) # Alternatively, do `len(knn_reg._y)`

print(offline_rss.shape)
print(offline_rss.__class__)
print(offline_location.shape) # Alternatively, do `len(offline_location)`
print(offline_location.__class__)
```

**Comment by PFloyd0** [comment]
<img width="384" alt="Capture3" src="https://user-images.githubusercontent.com/73516559/161511399-a13cb490-ede4-4ab0-b662-8025a85aa8bb.PNG">
Yes, they are raw numpy arrays. I load them from .mat file
Should I convert them to dataframe before training?
